ledger/internal/cow.go

```diff
@@ -317,6 +329,25 @@ func (cb *roundCowState) modifiedAccounts() []basics.Address {
 	return cb.mods.Accts.ModifiedAccounts()
 }
 
+// recycle resets the roundcowstate and returns it to the sync.Pool
+func (cb *roundCowState) recycle() {
+	cb.lookupParent = nil
+	cb.commitParent = nil
+	cb.proto = config.ConsensusParams{}
+	cb.mods.Reset()
+	cb.txnCount = 0
+	for addr := range cb.sdeltas {
+		delete(cb.sdeltas, addr)
+	}
+	cb.compatibilityMode = false
+	for addr := range cb.compatibilityGetKeyCache {
+		delete(cb.compatibilityGetKeyCache, addr)
+	}
+	cb.prevTotals = ledgercore.AccountTotals{}
+
+	childPool.Put(cb)
```

Comment:
Moved the resetting to happen right before Put as discussed, but not convinced that it's safer than resetting after calls to Get since some other code manually calling childPool.Put would yield a dirty object from Get